### PR TITLE
Allow Scene to report its total memory usage

### PR DESCRIPTION
**Pull request by mrobinson**
This is useful for memory reporting in Servo.

### CODE REVIEW

**Comment by mrobinson** [comment]
@metajack @nnethercote r?

File: src/tiling.rs [hunk 1]
```diff
@@ -334,4 +334,15 @@ impl TileGrid {
             tile.create_texture(display);
         }
     }
+
+    pub fn get_memory_usage(&self) -> usize {
+        self.tiles.values().map(|ref tile| {
+            // We cannot use Option::map_or here because rust will
+            // complain about moving out of borrowed content.
+            match tile.buffer {
+                Some(ref buffer) => buffer.get_mem(),
```
Comment (jdm) [inline src/tiling.rs]: `tile.buffer.as_ref().map_or(0, |buffer| buffer.get_mem())` probably does the trick, but it's debatable as to whether that's more readable.

Comment (mrobinson) [inline src/tiling.rs]: Yes, though this version is a bit more verbose, I think that I find it a bit more readable too. :+1:

**Comment by glennw** [comment]
r=me after rebase

**Comment by mrobinson** [comment]
The latest version of the PR should fix the merge conflicts.

**Comment by nnethercote** [comment]
Argh, I reviewed this yesterday and had several comments but they're not here. I must have failed to submit them.

**Comment by nnethercote** [comment]
My main comment was that it's completely unclear where this memory is located -- the heap?  mmap'd memory?  Unknown?  It would be good if the function names and/or comments made this clear.

**Comment by mrobinson** [comment]
@nnethercote In the latest version, I added comments explaining that the memory was either allocated on the heap or via GPU memory. I'm happy to fix it up though, if you think something else is required.

**Comment by mrobinson** [comment]
@nnethercote I couldn't figure out how to reply to them though, I think they were destroyed when I uploaded the new version of the patch.

**Comment by metajack** [comment]
@larsbergstrom How do we hook up reviewable to other repos? That would have helped here.

**Comment by nnethercote** [comment]
@mrobinson: so you did! Great, thank you.

**Comment by larsbergstrom** [comment]
@metajack I've enabled reviewable on all Servo org repos now.

**Pull request merged**